### PR TITLE
[1LP][RFR] Red Hat Updates to widgetastic

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Convert Red Hat tab to widgetastic, update tests to use new class
{{pytest: cfme/tests/configure/test_register_appliance.py}}
Some of test_rh_registration failed, but they passed locally.

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/configure/configuration/region_settings.py [hunk 1]
```diff
@@ -428,3 +431,347 @@ class MapTagsEdit(CFMENavigateStep):
...
+        self.set_default_rhsm_address = set_default_rhsm_address
+        self.set_default_repository = set_default_repository
+        Navigatable.__init__(self, appliance=appliance)
+
+    @property
+    def service_types(self):
+        return {
+            'rhsm': 'Red Hat Subscription Management',
+            'sat6': 'Red Hat Satellite 6'
+        }
```
Comment (jkrocil) [inline cfme/configure/configuration/region_settings.py]: No reason to keep this as a property. It can be defined as a class variable.

Comment (rlbabyuk) [inline cfme/configure/configuration/region_settings.py]: Done

**Comment by dajoRH** [comment]
Lint report for commit 50aa77f5a9fb57027cae9d99518ab247620b7e92:

`cfme/configure/configuration/region_settings.py`:
- :bangbang: Line 523:13: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 525:9: **E121** *continuation line under-indented for hanging indent*
- :bangbang: Line 548:5: **E303** *too many blank lines (2)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 08d22d877b0120e53c2f46aab1f7f13fb3feceac:

`cfme/configure/configuration/region_settings.py`:
- :bangbang: Line 523:13: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 525:9: **E121** *continuation line under-indented for hanging indent*
- :bangbang: Line 548:5: **E303** *too many blank lines (2)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/configure/configuration/region_settings.py [hunk 2]
```diff
@@ -428,3 +431,344 @@ class MapTagsEdit(CFMENavigateStep):
...
+            view.validate_button.click()
+
+        if cancel:
+            view.cancel_button.click()
+            flash_message = 'Edit of Customer Information was cancelled'
+        else:
+            view.save_button.click()
+            flash_message = 'Customer Information successfully saved'
+
+        view = self.create_view(RedHatUpdatesView)
```
Comment (quarckster) [inline cfme/configure/configuration/region_settings.py]: **REQUIRED**
```python
assert view.is_displayed
```

File: cfme/configure/configuration/region_settings.py [hunk 3]
```diff
@@ -428,3 +431,344 @@ class MapTagsEdit(CFMENavigateStep):
...
+
+    def is_registering(self, *appliance_names):
+        """ Check if at least one appliance is registering """
+        view = navigate_to(self, 'Details')
+        for appliance_name in appliance_names:
+            row = view.updates_table.row(appliance=appliance_name)
+            if row.last_message.text.lower() == 'registering':
+                return True
+            else:
+                return False
```
Comment (quarckster) [inline cfme/configure/configuration/region_settings.py]: **REQUIRED**
A logic mistake is here. If the first appliance is not registering other ones won't be checked.
```python
for appliance_name in appliance_names:
    row = view.updates_table.row(appliance=appliance_name)
    if row.last_message.text.lower() == 'registering':
        return True
else:
    return False
```

File: cfme/configure/configuration/region_settings.py [hunk 4]
```diff
@@ -428,3 +431,344 @@ class MapTagsEdit(CFMENavigateStep):
...
+    def is_registered(self, *appliance_names):
+        """ Check if each appliance is registered
+
+        Args:
+            appliance_names: Names of appliances to check; will check all if empty
+        """
+        view = navigate_to(self, 'Details')
+        for appliance_name in appliance_names:
+            row = view.updates_table.row(appliance=appliance_name)
+            if row.last_message.text.lower() == 'registered':
```
Comment (quarckster) [inline cfme/configure/configuration/region_settings.py]: **REQUIRED**
The same is here
```python
for appliance_name in appliance_names:
    row = view.updates_table.row(appliance=appliance_name)
    if row.last_message.text.lower() == 'registered':
        return True
else:
    return False
```

File: cfme/configure/configuration/region_settings.py [hunk 5]
```diff
@@ -428,3 +431,344 @@ class MapTagsEdit(CFMENavigateStep):
...
+
+        Args:
+            appliance_names: Names of appliances to get; will get all if empty
+        """
+        view = navigate_to(self, 'Details')
+        if appliance_names:
+            rows = list()
+            for row in view.updates_table.rows():
+                if row.appliance.text in appliance_names:
+                    rows.append(row)
```
Comment (quarckster) [inline cfme/configure/configuration/region_settings.py]: **REQURED**
Please use list comprehensions
```python
rows = [row for view.updates_table.rows() if row.appliance.text in appliance_names]
```

File: cfme/configure/configuration/region_settings.py (hunk 2 above)
Comment (rlbabyuk) [inline cfme/configure/configuration/region_settings.py]: done

File: cfme/configure/configuration/region_settings.py (hunk 4 above)
Comment (rlbabyuk) [inline cfme/configure/configuration/region_settings.py]: Done

File: cfme/configure/configuration/region_settings.py (hunk 5 above)
Comment (rlbabyuk) [inline cfme/configure/configuration/region_settings.py]: done

**Comment by quarckster** [comment]
LGTM, but please check test failures

**Comment by rlbabyuk** [comment]
I've rerun tests cases, all passed except test_rh_registration[rhsm-proxy_on], but locally passed

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 83b9b4269f968b3b23039e41d511499d4b56352a:

`cfme/configure/configuration/region_settings.py`:
- :bangbang: Line 501:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 512:101: **E501** *line too long (105 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**